jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -145,6 +151,12 @@ object XGBoost extends Serializable {
       } finally {
         Rabit.shutdown()
         watches.delete()
+        cacheDirName.foreach { name =>
+          val cacheDir = new File(name)
+          if (!cacheDir.listFiles().forall(_.delete()) || !cacheDir.delete()) {
+            throw new IllegalStateException(s"failed to delete $cacheDir")
+          }
+        }
```

Comment:
shouldn't the cache files be created under executor's working directory which would be cleaned by Spark automatically?

Reply:
I think `cacheDirName` is inside the working directory since it's a relative path. I would rather do the cleanup manually because otherwise, we could fail the container due to consuming all available disk space (for no good reason).

Reply:
We could reduce the length of the filename though, taskId should be unique enough.

Reply:
it's possible that the user will create multiple models in the same spark application, so I would like to keep stageId

Reply:
I think for cleaning, at least we should do in watches.delete() instead of leaking to main training logic

Reply:
The reason I did not include it in `watches.delete` is that `Watches` does not create the directory, so imo it shouldn't remove it. However, it can remove the individual files for the matrices. Sounds good?

Reply:
it's fine, watches didn't create DMatrix either but still cleaned it....